plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -526,4 +526,62 @@ public function test_wrap_image_in_picture_with_false_image_src(): void {
 
 		$this->assertSame( $image, $filtered_image );
 	}
+
+	/**
+	 * Test that images are not wrapped in picture element for unsupported contexts.
+	 *
+	 * @dataProvider data_provider_webp_uploads_wrap_image_in_picture_with_different_context
+	 *
+	 * @param string $context  The context to test.
+	 * @param bool   $expected Whether the image should be wrapped in a picture element.
+	 */
+	public function test_webp_uploads_wrap_image_in_picture_with_different_context( string $context, bool $expected ): void {
+		$image = wp_get_attachment_image(
+			self::$image_id,
+			'large',
+			false,
+			array(
+				'class' => 'wp-image-' . self::$image_id,
+				'alt'   => 'Green Leaves',
+			)
+		);
+
+		$this->opt_in_to_picture_element();
+		$filtered_image = apply_filters( 'wp_content_img_tag', $image, $context, self::$image_id );
+		if ( $expected ) {
+			$this->assertStringStartsWith( '<picture', $filtered_image );
```

Comment:
Better to use the HTML Tag Processor here.

```suggestion
			$p = new WP_HTML_Tag_Processor( $filtered_image );
			$this->assertTrue( $p->next_tag() );
			$this->assertSame( 'PICTURE', $p->get_tag() );
			$this->assertTrue( $p->next_tag() );
			$this->assertSame( 'SOURCE', $p->get_tag() );
			$this->assertTrue( $p->next_tag() );
			$this->assertSame( 'IMG', $p->get_tag() );
```